src/components/DataDimension/DndContext.js

```diff
@@ -0,0 +1,165 @@
+            onDragEnd={handleDragEnd}
+            onDragCancel={handleDragCancel}
+            sensors={sensors}
+        >
+            {children}
+            <DragOverlay dropAnimation={null}>
+                {draggingItem ? (
+                    <span className="dragOverlay">
+                        <DraggingItem item={draggingItem} />
+                    </span>
+                ) : null}
+            </DragOverlay>
+        </DndContext>
+    )
+
+    function handleDragStart({ active }) {
+        setDraggingItem(active.data.current)
+        onDragStart && onDragStart()
+    }
+
+    function handleDragCancel() {
+        setDraggingItem(null)
+    }
+
+    function handleDragEnd({ active, over }) {
+        if (
+            !over?.id ||
+            over?.data?.current?.sortable?.containerId === OPTIONS_PANEL
+        ) {
+            // dropped over non-droppable or over options panel
+            handleDragCancel()
+            return
+        }
+
+        const activeItem = {
+            id: active.id,
+            sourceAxisId: active.data.current?.sortable?.containerId,
+            sourceIndex: active.data.current?.sortable?.index,
+            data: {
+                id: active.data.current.id,
+                type: active.data.current.type,
+                label: active.data.current.label,
+            },
+        }
+        const overItem = {
+            axisId: over.data.current?.sortable?.containerId || over.id,
+            index: over.data.current?.sortable?.index,
+        }
+        onDragEnd({ activeItem, over: overItem })
+        setDraggingItem(null)
+    }
+
+    function getIntersectionRatio(entry, target) {
+        const top = Math.max(target.top, entry.top)
+        const left = Math.max(target.left, entry.left)
+        const right = Math.min(
+            target.left + target.width,
+            entry.left + entry.width
+        )
+        const bottom = Math.min(
+            target.top + target.height,
+            entry.top + entry.height
+        )
+        const width = right - left
+        const height = bottom - top
+
+        if (left < right && top < bottom) {
+            const targetArea = target.width * target.height
+            const entryArea = entry.width * entry.height
+            const intersectionArea = width * height
+            const intersectionRatio =
+                intersectionArea / (targetArea + entryArea - intersectionArea)
+            return Number(intersectionRatio.toFixed(4))
+        } // Rectangles do not overlap, or overlap has an area of zero (edge/corner overlap)
+
+        return 0
+    }
+    function sortCollisionsDesc(
+        { data: { value: a } },
+        { data: { value: b } }
+    ) {
+        return b - a
+    }
+
+    function rectIntersectionCustom({
+        pointerCoordinates,
+        droppableContainers,
+    }) {
+        // create a rect around the pointerCoords for calculating the intersection
+
+        const pointerRectWidth = 40
```

Comment:
I don't know where `40` is coming from, but if the width and height are equal then maybe just use a single prop instead of two? (e.g. `pointerRectSize`?)

Reply:
They happen to be the same, but they don't have to be. My goal is to refactor this function so it can be used by multiple dnd implementations, so I'll leave it as is.